src/main/java/nc/students/ayaz/Video.java

```diff
@@ -44,4 +24,19 @@ void addComment(Comment comment) {
     void deleteComment(Comment comment) {
         comments.remove(comment);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
It is auto generated code. (by IDEA)
That means they know what they are doing ))
But the actual reason, I think, is performance. Its much cheaper to compare references than comparing full objects.